DataFormats/PortableMath/interface/deltaPhi.h

```diff
@@ -0,0 +1,45 @@
+#ifndef DataFormats_PortableMath_deltaPhi_h
+#define DataFormats_PortableMath_deltaPhi_h
+
+/* functions to compute heterogeneous deltaPhi
+ * Ported from the LST algorithm:
+ * https://github.com/cms-sw/cmssw/pull/45117#discussion_r1749143439
+ */
+
+namespace cms::alpakatools {
+
+  // reduce to [-pi,pi]
+  template <typename TAcc, typename T>
+  ALPAKA_FN_HOST_ACC ALPAKA_FN_INLINE constexpr T reduceRange(TAcc const& acc, T x) {
+    constexpr T o2pi = 1. / (2. * M_PI);
+    if (alpaka::math::abs(acc, x) <= T(M_PI))
+      return x;
+    T n = alpaka::math::round(acc, x * o2pi);
+    return x - n * T(2. * M_PI);
+  }
+
+  template <typename TAcc, typename T>
+  ALPAKA_FN_HOST_ACC ALPAKA_FN_INLINE constexpr T phi(TAcc const& acc, T x, T y) {
+    return reduceRange(acc, M_PI + alpaka::math::atan2(acc, -y, -x));
+  }
+
+  template <typename TAcc, typename T>
+  ALPAKA_FN_HOST_ACC ALPAKA_FN_INLINE constexpr T deltaPhi(TAcc const& acc, T x1, T y1, T x2, T y2) {
+    T phi1 = phi(acc, x1, y1);
+    T phi2 = phi(acc, x2, y2);
+    return reduceRange(acc, phi2 - phi1);
+  }
```

Comment:
isn't it more precise and faster to `reduceRange(acc, alpaka::math::atan2(acc, -y2, -x2) - alpaka::math::atan2(acc, -y1, -x1))` ?

Reply:
Indeed, I will switch to that. The `phi()` function is needed elsewhere in the LST code, so I will keep it as well.